cli_tools/common/utils/storage/storage_object.go

```diff
@@ -0,0 +1,90 @@
+//  Copyright 2020 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package storage
+
+import (
+	"context"
+	"io"
+
+	"cloud.google.com/go/storage"
+	"github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/common/domain"
+)
+
+// storageObjectCreator is responsible for creating GCS Object.
+type storageObjectCreator struct {
+	ctx context.Context
+	sc  *storage.Client
+}
+
+// GetObject gets GCS object.
+func (ohc *storageObjectCreator) GetObject(
+	bucket string, object string) domain.StorageObject {
+	return &storageObject{
+		oh: ohc.sc.Bucket(bucket).Object(object), ctx: ohc.ctx}
+}
+
+// storageObject implements StorageObject.
+type storageObject struct {
+	oh  *storage.ObjectHandle
+	ctx context.Context
+}
+
+// GetObjectHandle gets the storage object handle.
+func (so *storageObject) GetObjectHandle() *storage.ObjectHandle {
+	return so.oh
+}
+
+// Delete deletes GCS object.
+func (so *storageObject) Delete() error {
+	return so.oh.Delete(so.ctx)
+}
+
+// NewReader creates a new Reader to read the contents of the object.
+func (so *storageObject) NewReader() (io.ReadCloser, error) {
+	return so.oh.NewReader(so.ctx)
+}
+
+// NewWriter creates a new Writer to write to the object.
+func (so *storageObject) NewWriter() io.WriteCloser {
+	return so.oh.NewWriter(so.ctx)
+}
+
+// ObjectName returns the name of the object.
+func (so *storageObject) ObjectName() string {
+	return so.oh.ObjectName()
+}
+
+/*
+Compose takes in srcs as source objects and concatenates them into the destination storage object.
+Returns the object attributes of the destination object after composed.
+Up to 32 objects can be composed into a one object.
+This function is used during uploading to GCS. To increase upload speed of a file,
+the file is split into multiple parts and concurrently uploaded to storage objects.
+At the end of the upload, the function is called to compose the objects back into one
+storage object.
```

Comment:
```suggestion
Compose takes in srcs as source objects and concatenates them into the destination storage object.
Returns the object attributes of the destination object after composed.
Up to 32 objects can be composed into a one object.
This function can be used to make uploads to GCS faster. To increase upload speed of a file,
the file can be split into multiple parts and concurrently uploaded to storage objects.
At the end of the upload, this function should be called to compose the objects back into one
storage object.
```

We should not state how clients are using a functions as that can change. Stating it as an example is ok.